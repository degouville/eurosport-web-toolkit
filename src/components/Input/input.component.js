import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import Eye from 'src/assets/eye.component.svg';

const INPUT_TEXT = 'text';
const INPUT_PASSWORD = 'password';

const Input = ({ placeholder, type, value, onChange, textButton, hasLabelInside }) => {
  const [inputType, setInputType] = useState(type);
  const [focused, setFocused] = useState(value !== '');

  const onBlur = useCallback(() => value === '' && setFocused(false), [value, setFocused]);
  const onFocus = useCallback(() => setFocused(true), [setFocused]);
  const onShowPasswordClick = useCallback(
    () => (inputType === INPUT_PASSWORD ? setInputType(INPUT_TEXT) : setInputType(INPUT_PASSWORD)),
    [inputType, setInputType]
  );

  const showPasswordEnabled = inputType === INPUT_PASSWORD;

  return (
    <>
      {!hasLabelInside && (
        <Placeholder focused={focused} hasLabelInside={hasLabelInside}>
          {placeholder}
        </Placeholder>
      )}
      <MainContainer focused={focused}>
        <InputContainer focused={focused} hasLabelInside={hasLabelInside}>
          {hasLabelInside && (
            <Placeholder focused={focused} hasLabelInside={hasLabelInside}>
              {placeholder}
            </Placeholder>
          )}
          <InputText
            type={inputType}
            focused={focused}
            value={value}
            onChange={onChange}
            onBlur={onBlur}
            onFocus={onFocus}
            hasLabelInside={hasLabelInside}
          />
        </InputContainer>
        {type === INPUT_PASSWORD && (
          <ShowPasswordContainer active={showPasswordEnabled} onClick={onShowPasswordClick}>
            <Text>{textButton}</Text>
            <Icon />
          </ShowPasswordContainer>
        )}
      </MainContainer>
    </>
  );
};

Input.defaultProps = {
  placeholder: '',
  type: INPUT_TEXT,
  textButton: '',
  hasLabelInside: false,
};

Input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  textButton: PropTypes.string,
  hasLabelInside: PropTypes.bool,
};

const Icon = styled(Eye)`
  margin-right: 16px;
  width: 24px;
  height: 24px;
  cursor: pointer;
  transition: all 250ms ease;
`;

const Text = styled.p`
  margin-right: 8px;
  margin-left: 8px;
  height: 14px;
  font-size: 12px;
  font-weight: 600;
  letter-spacing: 1px;
  line-height: 14px;
  text-align: right;
  text-transform: uppercase;
  cursor: pointer;
  transition: all 250ms ease;
`;

export const ShowPasswordContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  ${Icon} {
    fill: ${({ active, theme }) => (!active ? theme.input.borderFocused : theme.input.border)};
  }

  ${Text} {
    color: ${({ active, theme }) => (!active ? theme.input.borderFocused : theme.input.border)};
    font-family: ${({ theme }) => theme.input.placeholder.fontFamily};
  }

  &:hover ${Icon} {
    fill: ${({ theme }) => theme.input.borderFocused};
  }

  &:hover ${Text} {
    color: ${({ theme }) => theme.input.borderFocused};
  }

  transition: all 250ms ease;
`;

const MainContainer = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: 1px solid ${({ focused, theme }) => (focused ? theme.input.borderFocused : theme.input.border)};
  transition: all 250ms ease;
  margin-top: ${props => (props.hasLabelInside ? '0' : '13px')};
  border-radius: 4px;
  background-color: ${({ theme }) => theme.input.backgroundColor};
`;

const InputContainer = styled.div`
  ${props => (props.hasLabelInside ? 'height: 72px' : 'height: 48px')};
  display: flex;
  flex: 1;
  flex-direction: column;
  ${({ focused }) => !focused && `justify-content: center;`}
  transition: all 250ms ease;
`;

export const Placeholder = styled.label`
  position: ${props => (props.hasLabelInside ? 'absolute' : 'static')};
  user-select: none;
  margin-left: ${props => (props.hasLabelInside ? '24px' : '0')};
  margin-right: 24px;
  ${({ focused }) => focused && `margin-top: 18px;`}
  height: 14px;
  color: ${({ theme }) => theme.input.placeholder.color};
  font-family: ${({ theme }) => theme.input.placeholder.fontFamily};
  font-size: 12px;
  font-weight: 600;
  letter-spacing: 1px;
  line-height: 14px;
  text-transform: uppercase;
`;

export const InputText = styled.input`
  ${({ focused, hasLabelInside }) => focused && (hasLabelInside ? `margin-top: 34px;` : `margin-top: 12px;`)}
  position: relative;
  outline: 0;
  border: 0;
  border: none;
  padding: 0;
  text-decoration: none;
  background-color: transparent;
  height: 24px;
  color: ${({ theme }) => theme.input.text.color};
  font-family: ${({ theme }) => theme.input.text.fontFamily};
  font-size: 16px;
  line-height: 24px;

  // NOTE: FIX for android smartphone only
  width: 90%;
  margin: 10px;

  @media (min-width: 375px) {
    width: unset;
    margin: 0 24px 13px 24px;
  }
`;

export default Input;
