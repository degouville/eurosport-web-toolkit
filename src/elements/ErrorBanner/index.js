import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import errorIcon from 'src/assets/error-icon.svg';

const ErrorBanner = ({ message }) => (
  <MainContainer>
    <Border />
    <Icon src={errorIcon} alt="errorIcon" />
    <Text>{message}</Text>
  </MainContainer>
);

ErrorBanner.propTypes = {
  message: PropTypes.string.isRequired,
};

const MainContainer = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: flex-start;
  border: 1px solid ${({ theme }) => theme.banner.error.borderColor};
  background-color: ${({ theme }) => theme.banner.error.backgroundColor};
`;

const Border = styled.div`
  height: 100%;
  width: 5px;
  background-color: ${({ theme }) => theme.banner.error.borderColor};
`;

const Icon = styled.img`
  margin-top: 9px;
  margin-left: 17px;
  height: 24px;
  width: 24px;
`;

const Text = styled.p`
  color: ${({ theme }) => theme.banner.error.textColor};
  font-family: ${({ theme }) => theme.banner.error.fontFamily};
  font-size: 16px;
  line-height: 24px;
  margin: 9px 18px 15px 12px;
  flex: 1;
`;

export default ErrorBanner;
