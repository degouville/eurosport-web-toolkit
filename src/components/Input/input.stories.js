import React, { useState, useCallback } from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import Input from './input.component';

const indexStories = storiesOf('Components|Input', module);

const ControlledInput = props => {
  const [value, setValue] = useState('');
  const onChange = useCallback(event => setValue(event.target.value), [setValue]);
  return <Input {...props} value={value} onChange={onChange} />;
};

indexStories
  .add('text input', () => (
    <Input
      placeholder={text('placeholder', 'EMAIL ADDRESS')}
      type="text"
      value={text('value', 'test@eurosport.com')}
      onChange={() => null}
      hasLabelInside={boolean('hasLabelInside', true)}
    />
  ))
  .add('text input controlled', () => (
    <ControlledInput
      placeholder={text('placeholder', 'EMAIL ADDRESS')}
      type="text"
      hasLabelInside={boolean('hasLabelInside', true)}
    />
  ))
  .add('password input', () => (
    <Input
      placeholder={text('placeholder', 'SECRET PASSWORD')}
      type="password"
      value={text('value', 'secret')}
      onChange={() => null}
      textButton="SHOW"
      hasLabelInside={boolean('hasLabelInside', false)}
    />
  ))
  .add('password input controlled', () => (
    <ControlledInput
      placeholder={text('placeholder', 'SECRET PASSWORD')}
      type="password"
      textButton="SHOW"
      hasLabelInside={boolean('hasLabelInside', true)}
    />
  ));
