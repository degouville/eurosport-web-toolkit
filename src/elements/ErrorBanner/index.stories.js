import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import ErrorBanner from './index';

const indexStories = storiesOf('Elements|ErrorBanner', module);

indexStories.add('default', () => (
  <ErrorBanner message={text('message', 'You entered an incorrect email address or password, please try again')} />
));
