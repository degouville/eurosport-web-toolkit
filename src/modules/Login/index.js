import React, { useState, useCallback, useRef, useEffect } from 'react';
import styled, { css } from 'react-emotion';
import ReCAPTCHA from 'react-google-recaptcha';
import PropTypes from 'prop-types';
import Input from 'src/components/Input/input.component';
import ErrorBanner from 'src/elements/ErrorBanner';
import Button from 'src/elements/Button';
import ArrowLink from 'src/elements/ArrowLink';
import UnderlineLink from 'src/elements/UnderlineLink';
import { small, medium, large } from 'src/breakpoints';

const Login = ({
  onSubmit,
  forgotPasswordUrl,
  forgotPasswordText,
  subscribeUrl,
  subscribeText,
  needHelpText,
  needHelpUrl,
  callToActionText,
  passwordPlaceholder,
  emailPlaceholder,
  signInText,
  showPasswordText,
  title,
  errorMessage,
  showSubscribeSection,
  isSecondaryMode,
  recaptchaSiteKey,
}) => {
  const recaptchaRef = useRef(null);
  const emailRef = useRef(null);
  const passwordRef = useRef(null);

  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');

  useEffect(() => {
    emailRef.current = email;
    passwordRef.current = password;
  });

  const [isRecaptchaLoaded, setisRecaptchaLoaded] = useState(false);
  const [recaptchaToken, setRecaptchaToken] = useState(null);

  const handleSubmit = useCallback(
    event => {
      event && event.preventDefault();
      if (recaptchaToken !== null) {
        recaptchaRef.current.reset();
        setRecaptchaToken(null);
      }
      isRecaptchaLoaded && recaptchaRef.current.execute();
    },
    [recaptchaToken, isRecaptchaLoaded, recaptchaRef, setRecaptchaToken]
  );

  const onSignInSubmit = useCallback(() => handleSubmit(), [handleSubmit]);

  const onFormSubmit = useCallback(event => handleSubmit(event), [handleSubmit]);

  const onRecaptchaChange = useCallback(
    token => {
      setRecaptchaToken(token);
      if (token !== null) {
        onSubmit({ email: emailRef.current, password: passwordRef.current, recaptchaToken: token });
      }
    },
    [setRecaptchaToken, onSubmit]
  );

  const asyncRecaptchaOnLoad = useCallback(() => {
    setisRecaptchaLoaded(true);
  }, [setisRecaptchaLoaded]);

  const onEmailChange = useCallback(({ target }) => setEmail(target.value), [setEmail]);
  const onPasswordChange = useCallback(({ target }) => setPassword(target.value), [setPassword]);

  return (
    <FormContainer onSubmit={onFormSubmit}>
      <Title>{title}</Title>
      {errorMessage && (
        <ComponentContainer>
          <ErrorBanner message={errorMessage} />
        </ComponentContainer>
      )}
      <InputsContainer>
        <ComponentContainer>
          <Input value={email} onChange={onEmailChange} placeholder={emailPlaceholder} />
        </ComponentContainer>
        <ComponentContainer>
          <Input
            value={password}
            onChange={onPasswordChange}
            placeholder={passwordPlaceholder}
            type="password"
            textButton={showPasswordText}
          />
        </ComponentContainer>
        <ComponentContainer>
          <Button
            type={isSecondaryMode ? 'secondaryForm' : 'form'}
            onClick={onSignInSubmit}
            arrowType="arrow"
            tabIndex="0"
          >
            {signInText}
          </Button>
          <InvisibleSubmit type="submit" />
          {recaptchaSiteKey && (
            <ReCAPTCHA
              ref={recaptchaRef}
              size="invisible"
              sitekey={recaptchaSiteKey}
              theme="dark"
              onChange={onRecaptchaChange}
              asyncScriptOnLoad={asyncRecaptchaOnLoad}
            />
          )}
        </ComponentContainer>
      </InputsContainer>
      <ActionLinkContainer>
        <UnderlineLink href={forgotPasswordUrl}>{forgotPasswordText}</UnderlineLink>
        <LinkSeparator />
        <UnderlineLink href={needHelpUrl}>{needHelpText}</UnderlineLink>
      </ActionLinkContainer>
      {showSubscribeSection && (
        <>
          <Separator />
          <CallToActionContainer>
            <Text>{callToActionText}</Text>
            <ArrowLink href={subscribeUrl}>{subscribeText}</ArrowLink>
          </CallToActionContainer>
        </>
      )}
    </FormContainer>
  );
};

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  forgotPasswordText: PropTypes.string.isRequired,
  forgotPasswordUrl: PropTypes.string.isRequired,
  subscribeText: PropTypes.string.isRequired,
  subscribeUrl: PropTypes.string.isRequired,
  needHelpText: PropTypes.string.isRequired,
  needHelpUrl: PropTypes.string.isRequired,
  callToActionText: PropTypes.string.isRequired,
  passwordPlaceholder: PropTypes.string.isRequired,
  emailPlaceholder: PropTypes.string.isRequired,
  signInText: PropTypes.string.isRequired,
  showPasswordText: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  errorMessage: PropTypes.string,
  showSubscribeSection: PropTypes.bool,
  isSecondaryMode: PropTypes.bool,
  recaptchaSiteKey: PropTypes.string,
};

Login.defaultProps = {
  errorMessage: undefined,
  showSubscribeSection: true,
  isSecondaryMode: false,
  recaptchaSiteKey: '',
};

export const FormContainer = styled.form`
  display: flex;
  flex-direction: column;
  flex: 1;
  ${large(css`
    max-width: 500px;
  `)};
`;

const ComponentContainer = styled.div`
  margin-top: 28px;
`;

const InputsContainer = styled.div`
  ${medium(css`
    display: grid;
    grid-auto-columns: 1fr;
    grid-auto-flow: column;
    grid-column-gap: 16px;
    align-items: end;
  `)}
  ${large(css`
    display: block;
  `)};
`;

const ActionLinkContainer = styled.div`
  margin-top: 28px;
  display: flex;
  flex-direction: row;
  align-items: center;
  flex: 1;
`;

const LinkSeparator = styled.div`
  box-sizing: border-box;
  margin-left: 16px;
  margin-right: 16px;
  width: 1px;
  height: 24px;
  border: 1px solid ${({ theme }) => theme.login.separator.secondary};
`;

const Separator = styled.div`
  height: 1px;
  margin-top: 32px;
  margin-bottom: 32px;
  background-color: ${({ theme }) => theme.login.separator.primary};
  opacity: 0.15;
`;

export const CallToActionContainer = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
`;

const Text = styled.p`
  height: 24px;
  color: ${({ theme }) => theme.login.text.color};
  ${({ theme }) => theme.login.text.fontFamily}
  font-size: 16px;
  line-height: 24px;
  margin-right: 8px;
`;

const Title = styled.h1`
  color: ${({ theme }) => theme.login.title.color};
  ${({ theme }) => theme.login.title.fontFamily}
  font-size: 24px;
  line-height: 32px;
  ${small(css`
    font-size: 30px;
  `)};
  ${medium(css`
    line-height: 36px;
  `)};
`;

export const InvisibleSubmit = styled.input`
  display: none;
`;

export default Login;
