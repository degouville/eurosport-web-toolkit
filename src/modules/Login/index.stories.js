import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Login from './index';

const createDefaultProps = () => ({
  onSubmit: action('onSubmit'),
  title: text('title', 'Sign in to your Eurosport Player account'),
  forgotPasswordText: text('forgotPasswordText', 'Forgot password?'),
  forgotPasswordUrl: text('forgotPasswordUrl', 'https://eurosport.fr'),
  subscribeText: text('subscribeText', 'Subscribe'),
  subscribeUrl: text('subscribeUrl', 'https://eurosport.fr'),
  needHelpText: text('needHelpText', 'Need help?'),
  needHelpUrl: text('needHelpUrl', 'https://eurosport.fr'),
  callToActionText: text('callToActionText', 'New to Eurosport Player?'),
  passwordPlaceholder: text('passwordPlaceholder', 'Password'),
  emailPlaceholder: text('emailPlaceholder', 'Email'),
  signInText: text('signInText', 'Sign In'),
  showPasswordText: text('showPasswordText', 'Show'),
  showSubscribeSection: boolean('showSubscribeSection', true),
  isSecondaryMode: boolean('isSecondaryMode', false),
  recaptchaSiteKey: text('recaptchaSiteKey', '6LeHaK0UAAAAAF5SxiYjhFNMDpTKy4B8Mc_kWgiM'),
});

const createPropsWithErrorMessage = () => ({
  ...createDefaultProps(),
  errorMessage: text('errorMessage', 'Impossible to connect to your account'),
});

storiesOf('Modules|Login', module)
  .add('Desktop without error', () => (
    <div style={{ width: '417px' }}>
      <Login {...createDefaultProps()} />
    </div>
  ))
  .add('Desktop with error', () => (
    <div style={{ width: '417px' }}>
      <Login {...createPropsWithErrorMessage()} />
    </div>
  ))
  .add('Mobile without error', () => <Login {...createDefaultProps()} />)
  .add('Mobile with error', () => <Login {...createPropsWithErrorMessage()} />);
