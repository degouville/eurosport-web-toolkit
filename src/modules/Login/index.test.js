import React from 'react';
import { mount, shallow } from 'enzyme';
import ReCAPTCHA from 'react-google-recaptcha';
import { ThemeProvider } from 'emotion-theming';
import { act } from 'react-dom/test-utils';
import Login, { CallToActionContainer, FormContainer } from './index';
import ErrorBanner from '../../elements/ErrorBanner';
import Input from '../../components/Input/input.component';
import { theme } from '../..';

describe('Login', () => {
  const createProps = newProps => ({
    onSubmit: jest.fn(),
    title: 'Sign In',
    forgotPasswordUrl: 'es.com/forgot',
    forgotPasswordText: 'Forgot password',
    subscribeUrl: 'es.com/subscribe',
    subscribeText: 'Subscribe',
    needHelpText: 'Need Help?',
    needHelpUrl: 'es.com/help',
    callToActionText: 'New to eurosport player?',
    passwordPlaceholder: 'Password',
    emailPlaceholder: 'Email',
    signInText: 'SignIn',
    showPasswordText: 'Show',
    errorMessage: undefined,
    ...newProps,
  });

  describe('Snapshot testing', () => {
    it('renders Login without ErrorBanner', () => {
      // Given
      const props = createProps({ errorMessage: undefined });
      const wrapper = shallow(<Login {...props} />);

      // Expect
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('renders Login with ErrorBanner', () => {
    // Given
    const props = createProps({ errorMessage: 'Impossible to connect' });
    const wrapper = shallow(<Login {...props} />);

    // Expect
    expect(wrapper.find(ErrorBanner)).toHaveLength(1);
  });

  it('renders Login without subscription section', () => {
    // Given
    const props = createProps({ showSubscribeSection: false });
    const wrapper = shallow(<Login {...props} />);

    // Expect
    expect(wrapper.find(CallToActionContainer)).toHaveLength(0);
  });

  describe('Behaviour testing', () => {
    const getEmailInput = wrp => wrp.find(Input).find({ placeholder: 'Email' });
    const getPasswordInput = wrp => wrp.find(Input).find({ type: 'password' });
    const getFormContainer = wrp => wrp.find(FormContainer);

    it('Should not display ErrorBanner when it is not defined', () => {
      // Given
      const props = createProps({ errorMessage: undefined });
      const wrapper = shallow(<Login {...props} />);

      // When
      const elements = wrapper.find(ErrorBanner);

      // Expect
      expect(elements.length).toBe(0);
    });

    it('Should display ErrorBanner when it is defined', () => {
      // Given
      const props = createProps({ errorMessage: 'Impossible to connect' });
      const wrapper = shallow(<Login {...props} />);

      // When
      const elements = wrapper.find(ErrorBanner);

      // Expect
      expect(elements.length).toBe(1);
    });

    it('Should update mail value when user write his email', () => {
      // Given
      const props = createProps();
      const wrapper = shallow(<Login {...props} />);

      // When
      const onEmailChange = getEmailInput(wrapper).prop('onChange');
      onEmailChange({ target: { value: 'myEmail' } });

      wrapper.rerender();
      const email = getEmailInput(wrapper).prop('value');

      // Expect
      expect(email).toBe('myEmail');
    });

    it('Should update password value when user write his password', () => {
      // Given
      const props = createProps();
      const wrapper = shallow(<Login {...props} />);

      // When
      const onPasswordChange = getPasswordInput(wrapper).prop('onChange');
      onPasswordChange({ target: { value: 'secret' } });

      wrapper.rerender();
      const password = getPasswordInput(wrapper).prop('value');

      // Expect
      expect(password).toBe('secret');
    });

    it('Should reset reCAPTCHA on form submit', () => {
      // Given
      const props = createProps();
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <Login {...props} recaptchaSiteKey="site-key" />
        </ThemeProvider>
      );
      const reCAPTCHA = wrapper.find(ReCAPTCHA);
      const resetCallback = jest.fn();

      reCAPTCHA.getElement().ref.current.reset = resetCallback;

      // When
      act(() => {
        reCAPTCHA.props().onChange('token');
      });
      act(() => {
        getFormContainer(wrapper).simulate('submit');
      });

      // Expect
      expect(resetCallback).toHaveBeenCalled();
      wrapper.unmount();
    });
  });
});
