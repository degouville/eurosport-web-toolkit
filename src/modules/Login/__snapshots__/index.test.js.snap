// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot testing renders Login without ErrorBanner 1`] = `
<Styled(form)
  onSubmit={[Function]}
>
  <Styled(h1)>
    Sign In
  </Styled(h1)>
  <Styled(div)>
    <Styled(div)>
      <Input
        hasLabelInside={false}
        onChange={[Function]}
        placeholder="Email"
        textButton=""
        type="text"
        value=""
      />
    </Styled(div)>
    <Styled(div)>
      <Input
        hasLabelInside={false}
        onChange={[Function]}
        placeholder="Password"
        textButton="Show"
        type="password"
        value=""
      />
    </Styled(div)>
    <Styled(div)>
      <Button
        arrowType="arrow"
        onClick={[Function]}
        tabIndex="0"
        type="form"
      >
        SignIn
      </Button>
      <Styled(input)
        type="submit"
      />
    </Styled(div)>
  </Styled(div)>
  <Styled(div)>
    <UnderlineLink
      href="es.com/forgot"
    >
      Forgot password
    </UnderlineLink>
    <Styled(div) />
    <UnderlineLink
      href="es.com/help"
    >
      Need Help?
    </UnderlineLink>
  </Styled(div)>
  <Styled(div) />
  <Styled(div)>
    <Styled(p)>
      New to eurosport player?
    </Styled(p)>
    <ArrowLink
      arrowType="chevron"
      href="es.com/subscribe"
    >
      Subscribe
    </ArrowLink>
  </Styled(div)>
</Styled(form)>
`;
